pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -1755,6 +1759,47 @@ func createRewriteFilter(filter *k8s.HTTPURLRewriteFilter) *istio.HTTPRewrite {
 	return rewrite
 }
 
+func createCorsFilter(filter *k8s.HTTPCORSFilter) *istio.CorsPolicy {
+	if filter == nil {
+		return nil
+	}
+	res := &istio.CorsPolicy{}
+	for _, r := range filter.AllowOrigins {
+		rs := string(r)
+		if len(rs) == 0 {
+			continue // Not valid anyways, but double check
+		}
+		if strings.Contains(rs, "*") {
+			// Istio doesn't have suffix match for some reason...
+			regex := ".*" + regexp.QuoteMeta(rs[1:])
```

Comment:
My reading of the [GEP](https://gateway-api.sigs.k8s.io/geps/gep-1767/) doesn't seem to imply any restrictions on where the wildcard is applied, so assuming the RFC itself has no extra requirements, this doesn't seem to handle an input string of `foo*.com` correctly. AIIUC, this code would translate this to a regex of `.*oo.com` which would incorrectly match `btoo.com`

Reply:
Also, I initially thought `regexp.QuoteMeta(rs[1:])` if `rs == "*"` but then I remembered Go allows a[len(a):] to return an empty slice rather than an out-of-bounds error. Not sure if this was intentional or not, but I'd add a comment just for readability purposes

Reply:
currently the spec denies any wildcard. I put a request to clarify this https://github.com/kubernetes-sigs/gateway-api/issues/3648#issuecomment-2762935429

maybe safest to do no wildcard in the meantime..

Reply:
Yeah if the validation prevents wildcards, better to not implement IMO

Reply:
ok, reverted for now